File: Arrays and Hashing/encode-decode-strings.js
function encode(strs) {
  let encoded = "";
  for (let str of strs) {
    encoded += `${str.length}#${str}`;
  }
  return encoded;
}

function decode(str) {
  console.log(str);
  let i = 0;
  result = [];
  while (i < str.length) {
    let j = i;
    while (str[j] != "#") {
      j++;
    }

    let length = +str.slice(i, j);
    result.push(str.slice(j + 1, j + 1 + length));
    i = j + 1 + length;
  }
  return result;
}

File: Arrays and Hashing/encode-decode-strings.py
class Solution:
    """
    @param: strs: a list of strings
    @return: encodes a list of strings to a single string.
    """

    def encode(self, strs):
        # write your code here
        encoded_str = ''
        for s in strs:
            encoded_str += str(len(s)) + '#' + s
        return encoded_str

    """
    @param: str: A string
    @return: dcodes a single string to a list of strings
    """
    # 4#abcd

    def decode(self, str):
        i = 0
        res = []
        while i < len(str):
            j = i
            while str[j] != '#':
                j += 1
            length = int(str[i:j])
            res.append(str[j + 1: j + 1 + length])
            i = j + 1 + length
        return res
